sycl/test/filter_selector/select_device_number.cpp

```diff
@@ -0,0 +1,51 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RU: env SYCL_DEVICE_FILTER=2 %t.out
```

Comment:
Typo RU->RUN

So, which device we expect to see if SYCL_DEVICE_FILTER=2 considering given requirements? is it cpu device?
Is this behavior expected to be consistent on any system and in case of any device order in the platform?

Reply:
Nice catch of typo. 👍 
The relative index '2' really depends on the machine available devices list.
When I tested with requirements: opencl, cpu, gpu, accelerator, host, I found index '2' was the CPU device.
If we change the order of platforms in get_devices(), the index would change.
That''s why I did not implement this test case before.
But, since you asked, I added this test case and it should work for the current requirement.

Reply:
That's why I asked if it is possible to create a test for this. If behavior of the test is unpredictable and it could fail in case of different system configuration then let's better don't have it.

Reply:
okay, then let's remove this test.

Reply:
Done. I also had to fix typos in other test files.